include/rocksdb/configurable.h

```diff
@@ -0,0 +1,493 @@
+
+
+#pragma once
+
+#include <string>
+#include <unordered_map>
+#include <unordered_set>
+
+#include "rocksdb/status.h"
+#include "rocksdb/utilities/options_type.h"
+
+namespace rocksdb {
+struct DBOptions;
+struct ColumnFamilyOptions;
+class Logger;
+struct OptionTypeInfo;
+typedef std::unordered_map<std::string, OptionTypeInfo> OptionTypeMap;
+
+/**
+ * Configurable is an is an base class used by the rocksdb that describes a
+ * standard way of configuring objects.  A Configurable object can:
+ *   -> Populate itself given:
+ *        - One or more "name/value" pair strings
+ *        - A string repesenting the set of name=value properties
+ *        - A map of name/value properties.
+ *   -> Convert itself into its string representation
+ *   -> Dump itself to a Logger
+ *   -> Compare itself to another Configurable object to see if the two objects
+ * are equivalent
+ *
+ * If a derived class calls RegisterOptionsMap to register (by name) how its
+ * options objects are to be processed, this functionality can typically be
+ * handled by this class without additional overrides. Otherwise, the derived
+ * class will need to implement the methods for handling the corresponding
+ * functionality.
+ *
+ * The Configurable class also provides hooks to validate the option values
+ * associated with it
+ *   - The SanitizeOptions method can be used to check and change any options
+ * required for validity
+ *   - The ValidateOptions method can be used to check (but not change the
+ * values) of the options. These two methods must be overridden by derived
+ * classes to implement class-specific validation.
+ */
+class Configurable {
```

Comment:
According to https://google.github.io/styleguide/cppguide.html#Declaration_Order, can we re-organize the layout of the members of `Configurable`?